File: web-server/src/test/java/io/tipsters/error/RestErrorHandlerTest.java
package io.tipsters.error;

import org.junit.Test;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class RestErrorHandlerTest {

    private final RestErrorHandler underTest = new RestErrorHandler();

    @Test
    public void mapsTeamNotFoundError() throws Exception {
        TeamNotFoundError notFoundError = new TeamNotFoundError("Team ABC not found");

        ErrorResponse errorResponse = underTest.handleException(notFoundError);

        assertThat(errorResponse.getMessage(), is("Team ABC not found"));
        assertThat(errorResponse.getType(), is("TeamNotFoundError"));
    }
}

File: web-server/src/test/kotlin/io/tipsters/error/RestErrorHandlerTest.kt
package io.tipsters.error

import org.junit.Assert.assertThat
import org.junit.Test
import org.hamcrest.CoreMatchers.`is` as eq

class RestErrorHandlerTest {

    private val underTest = ErrorMapper()

    @Test
    fun mapsTeamNotFoundError() {
        val error = OddsApiError("Some error")

        val errorResponse = underTest.handleException(error)

        assertThat(errorResponse.message, eq("Some error"))
        assertThat(errorResponse.type, eq("OddsApiError"))
    }
}